include/nbl/asset/IAsset.h

```diff
@@ -210,6 +209,13 @@ class IAsset : virtual public core::IReferenceCounted
 
 		virtual bool canBeRestoredFrom(const IAsset* _other) const = 0;
 
+		//! 
+		virtual size_t hash() const = 0;
+
+		//!
+		virtual bool equals(const IAsset* _other) const = 0;
```

Comment:
you want to run basic checks here (like `_other` being the same asset type, etc.) and use a `protected virtual equals_impl` for the details

you probably want to do a similar refactor to `canBeRestoredFrom`